source/macos/AdaptiveCards/AdaptiveCards/Renderers/InputNumberRenderer.swift

```diff
@@ -66,18 +72,30 @@ open class ACRNumericTextField: NSView, NSTextFieldDelegate {
     }
     
     public func controlTextDidChange(_ obj: Notification) {
-        guard let textView = obj.object as? NSTextField else { return }
-        // updating the stepper value when text field value change
-        stepper.doubleValue = Double(textView.accessibilityValue() ?? "") ?? stepper.doubleValue
-        guard let stringValue = textView.accessibilityValue() else { return }
-        if Double(stringValue) != nil {
-            textView.stringValue = stringValue
-        } else if Int(stringValue) == nil && !stringValue.isEmpty {
-            textView.stringValue = previousValue
-        } else if stringValue.isEmpty {
-            textView.stringValue = ""
+        if let textfield = obj.object as? ACRTextField {
+            var stringValue = textfield.stringValue
+            
+            stepper.doubleValue = Double(textField.accessibilityValue() ?? "") ?? stepper.doubleValue
+            let charSet = NSCharacterSet(charactersIn: "1234567890.").inverted
```

Comment:
```suggestion
            stepper.doubleValue = Double(textField.stringValue ?? "") ?? stepper.doubleValue
            let charSet = NSCharacterSet(charactersIn: "1234567890.").inverted
```
`accessibilityValue()` might different purposes to solve